backend/src/v2/driver/util.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2021-2024 The Kubeflow Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package driver
+
+import (
+	"fmt"
+	"github.com/kubeflow/pipelines/api/v2alpha1/go/pipelinespec"
+	"regexp"
+)
+
+// InputPipelineChannelPattern define a regex pattern to match the content within single quotes
+// example input channel looks like "{{$.inputs.parameters['pipelinechannel--val']}}"
+const InputPipelineChannelPattern = `\$.inputs.parameters\['(.+?)'\]`
+
+func isInputParameterChannel(inputChannel string) bool {
+	re := regexp.MustCompile(InputPipelineChannelPattern)
+	match := re.FindStringSubmatch(inputChannel)
+	if len(match) == 2 {
+		return true
+	} else {
+		// if len(match) > 2, then this is still incorrect because
+		// inputChannel should contain only one parameter channel input
+		return false
+	}
+}
+
+// extractInputParameterFromChannel takes an inputChannel that adheres to
+// InputPipelineChannelPattern and extracts the channel parameter name.
+// For example given an input channel of the form "{{$.inputs.parameters['pipelinechannel--val']}}"
+// the channel parameter name "pipelinechannel--val" is returned.
+func extractInputParameterFromChannel(inputChannel string) (string, error) {
+	re := regexp.MustCompile(InputPipelineChannelPattern)
+	match := re.FindStringSubmatch(inputChannel)
+	if len(match) > 1 {
+		extractedValue := match[1]
+		return extractedValue, nil
+	} else {
+		return "", fmt.Errorf("failed to extract input parameter from channel: %s", inputChannel)
+	}
+}
+
+// resolvePodSpecRuntimeParameter resolves runtime value that is intended to be
+// utilized within the Pod Spec. parameterValue takes the form of:
+// "{{$.inputs.parameters['pipelinechannel--someParameterName']}}"
+//
+// parameterValue is a runtime parameter value that has been resolved and included within
+// the executor input. Since the pod spec patch cannot dynamically update the underlying
+// container template's inputs in an Argo Workflow, this is a workaround for resolving
+// such parameters.
+//
+// If parameter value is not a parameter channel, then a constant value is assumed and
+// returned as is.
+func resolvePodSpecRuntimeParameter(parameterValue string, executorInput *pipelinespec.ExecutorInput) (string, error) {
+	if isInputParameterChannel(parameterValue) {
+		inputImage, err1 := extractInputParameterFromChannel(parameterValue)
```

Comment:
I believe this methods can be used to query the value for container.Image and also other dynamic parameters. Consider using another name instead of inputImage

Reply:
It's actually only input runtime parameters, I'll update the parent function `resolvePodSpecRuntimeParameter` to reflect this